src/pybind/mgr/dashboard/ci/cephadm/start-cluster.sh

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+set -ex
+
+cleanup() {
+    set +x
+    if [[ -n "$JENKINS_HOME" ]]; then
+        printf "\n\nStarting cleanup...\n\n"
+        kcli delete plan -y ceph || true
+        docker container prune -f
+        printf "\n\nCleanup completed.\n\n"
+    fi
+}
+
+on_error() {
+    set +x
+    if [ "$1" != "0" ]; then
+        printf "\n\nERROR $1 thrown on line $2\n\n"
+        printf "\n\nCollecting info...\n\n"
+        for vm_id in 0 1 2
+        do
+            local vm="ceph-node-0${vm_id}"
+            printf "\n\nDisplaying journalctl from VM ${vm}:\n\n"
+            kcli ssh -u root -- ${vm} 'journalctl --no-tail --no-pager -t cloud-init' || true
+            printf "\n\nEnd of journalctl from VM ${vm}\n\n"
+            printf "\n\nDisplaying container logs:\n\n"
+            kcli ssh -u root -- ${vm} 'podman logs --names --since 30s $(podman ps -aq)' || true
+        done
+        printf "\n\nTEST FAILED.\n\n"
+    fi
+}
+
+trap 'on_error $? $LINENO' ERR
+trap 'cleanup $? $LINENO' EXIT
+
+sed -i '/ceph-node-/d' $HOME/.ssh/known_hosts
+
+: ${CEPH_DEV_FOLDER:=${PWD}}
+EXTRA_PARAMS=''
+DEV_MODE=''
+# Check script args/options.
+for arg in "$@"; do
+  shift
+  case "$arg" in
+    "--dev-mode") DEV_MODE='true'; EXTRA_PARAMS="-P dev_mode=${DEV_MODE}" ;;
+  esac
+done
+
+kcli delete plan -y ceph || true
```

Comment:
why not call to `cleanup`?

Reply:
> why not call to `cleanup`?

The `cleanup` function is mainly intended to do  cleanup on jenkins environment when the job finishes (on success or on error).
As this script can be run locally, I prefer to explicitly delete the plan and let the user decide - once the script finishes - when to delete plan (if you are running it locally probably you are in "developer mode" so better not to delete the plan automatically).